src/distribution.cpp

```diff
@@ -29,27 +29,67 @@ Discrete::Discrete(pugi::xml_node node)
   std::copy(params.begin(), params.begin() + n / 2, std::back_inserter(x_));
   std::copy(params.begin() + n / 2, params.end(), std::back_inserter(p_));
 
-  normalize();
+  // Initialize alias tables
+  init_alias();
 }
 
 Discrete::Discrete(const double* x, const double* p, int n)
   : x_ {x, x + n}, p_ {p, p + n}
 {
+  // Initialize alias tables
+  init_alias();
```

Comment:
I would move the `init_alias` method inline here in the constructor and not immediately set `p_` (or, rather, `prob_`) above.